src/main/java/io/stargate/sgv2/jsonapi/service/embedding/operation/AzureOpenAiEmbeddingClient.java

```diff
@@ -0,0 +1,111 @@
+package io.stargate.sgv2.jsonapi.service.embedding.operation;
+
+import com.fasterxml.jackson.annotation.JsonInclude;
+import io.quarkus.rest.client.reactive.ClientExceptionMapper;
+import io.quarkus.rest.client.reactive.QuarkusRestClientBuilder;
+import io.smallrye.mutiny.Uni;
+import io.stargate.sgv2.jsonapi.exception.ErrorCode;
+import io.stargate.sgv2.jsonapi.exception.JsonApiException;
+import io.stargate.sgv2.jsonapi.service.embedding.configuration.EmbeddingProviderConfigStore;
+import io.stargate.sgv2.jsonapi.service.embedding.configuration.EmbeddingProviderResponseValidation;
+import io.stargate.sgv2.jsonapi.service.embedding.operation.error.HttpResponseErrorMessageMapper;
+import io.stargate.sgv2.jsonapi.service.embedding.util.EmbeddingUtil;
+import jakarta.ws.rs.HeaderParam;
+import jakarta.ws.rs.POST;
+import jakarta.ws.rs.core.Response;
+import java.net.URI;
+import java.time.Duration;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.TimeUnit;
+import org.eclipse.microprofile.rest.client.annotation.ClientHeaderParam;
+import org.eclipse.microprofile.rest.client.annotation.RegisterProvider;
+import org.eclipse.microprofile.rest.client.inject.RegisterRestClient;
+
+public class AzureOpenAiEmbeddingClient implements EmbeddingProvider {
+  private EmbeddingProviderConfigStore.RequestProperties requestProperties;
+  private String modelName;
+  private int dimension;
+  private final OpenAiEmbeddingProvider embeddingProvider;
+
+  public AzureOpenAiEmbeddingClient(
```

Comment:
Can you add the docs?

Reply:
I did not see Javadocs for other clients, so not sure what to describe here? Link to Azure OpenAI docs maybe?

Reply:
Adding link from class Javadoc.